cglib/src/main/java/net/sf/cglib/core/TypeUtils.java

```diff
@@ -356,21 +359,21 @@ public static int LCONST(long value) {
     }
 
     public static int FCONST(float value) {
-        if (value == 0f) {
+        if (floatEquals(value, 0f)) {
             return Constants.FCONST_0;
-        } else if (value == 1f) {
+        } else if (floatEquals(value, 1f)) {
             return Constants.FCONST_1;
-        } else if (value == 2f) {
+        } else if (floatEquals(value, 2f)) {
             return Constants.FCONST_2;
         } else {
             return -1; // error
         }
     }
 
     public static int DCONST(double value) {
-        if (value == 0d) {
+        if (doubleEquals(value, 0)) {
             return Constants.DCONST_0;
-        } else if (value == 1d) {
+        } else if (doubleEquals(value, 1)) {
```

Comment:
@vlsi I think we don't know what is the `value` variable can contains, and according to this rule: [](https://dev.eclipse.org/sonar/rules/show/squid:S1244) , equality can lead to unexpected results.